streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -83,9 +84,27 @@ public void register(final StateRestorer restorer) {
 
         final Set<TopicPartition> restoringPartitions = new HashSet<>(needsRestoring.keySet());
         try {
-            final ConsumerRecords<byte[], byte[]> allRecords = poll(restoreConsumer, 10);
-            for (final TopicPartition partition : restoringPartitions) {
-                restorePartition(allRecords, partition, active.restoringTaskFor(partition));
+            updatedEndOffsets = !needsRestoring.isEmpty() ? 
```

Comment:
This would update `updateEndOffsets` multiple time -- should we set it only once? Note, that `needsRestoring` will not be empty until the full restore is completed.